### PR TITLE
feat: New Module Microsoft.Fabric/capacities - `avm/res/fabric/capacity`

### PR DESCRIPTION
**Pull request by hundredacres**
## Description

New bicep module to manage the Azure Fabric Capacity resource.

## Pipeline Reference

| Pipeline |
| -------- |
| [avm.res.fabric.capacity](https://github.com/hundredacres/bicep-registry-modules/actions/workflows/avm.res.fabric.capacity.yml)|

## Type of Change

- [ ] Update to CI Environment or utilities (Non-module affecting changes)
- [X] Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [X] I'm sure there are no other open Pull Requests for the same update/change
- [X] I have run `Set-AVMModule` locally to generate the supporting module files.
- [X] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

**Comment by eriqua** [comment]
@matebarabas is this module onboarded to MAR?

File: avm/res/network/front-door/ORPHANED.md
Comment (eriqua) [inline avm/res/network/front-door/ORPHANED.md:1]: Please remove this change as not related

File: avm/res/network/front-door/README.md
Comment (eriqua) [inline avm/res/network/front-door/README.md:1]: Please remove this change as not related

File: avm/res/fabric/capacity/main.bicep [hunk 1]
```diff
@@ -0,0 +1,110 @@
+metadata name = 'Fabric Capacities'
+metadata description = '<Add description>'
```
Comment (eriqua) [inline avm/res/fabric/capacity/main.bicep:2]: Please add description

File: avm/res/fabric/capacity/README.md [hunk 2]
```diff
@@ -0,0 +1,242 @@
+# Fabric Capacities `[Microsoft.Fabric/capacities]`
+
+<Add description>
```
Comment (eriqua) [inline avm/res/fabric/capacity/README.md:3]: Readme to be regenerate as a last step after addressing other comments

File: avm/res/fabric/capacity/main.bicep [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+  'F256'
+  'F512'
+  'F1024'
+  'F2048'
+])
+@description('Optional. SKU tier of the Fabric resource.')
+param skuName string = 'F2'
+
+@description('Optional. SKU name of the Fabric resource.')
+param skuTier string = 'Fabric'
```
Comment (eriqua) [inline avm/res/fabric/capacity/main.bicep:31]: Should this also have allowed values?

File: avm/res/fabric/capacity/main.bicep [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+@description('Optional. List of admin members. Format: ["something@domain.com"].')
+param adminMembers array = []
+
+@description('Optional. Enable/Disable usage telemetry for module.')
+param enableTelemetry bool = true
+
+//
+// Add your parameters here
```
Comment (eriqua) [inline avm/res/fabric/capacity/main.bicep:40]: Please remove comments, these are only template guidelines intended to be removed

File: avm/res/fabric/capacity/main.bicep [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+          type: 'String'
+          value: 'For more information, see https://aka.ms/avm/TelemetryInfo'
+        }
+      }
+    }
+  }
+}
+
+//
+// Add your resources here
```
Comment (eriqua) [inline avm/res/fabric/capacity/main.bicep:67]: Same here please remove comments

File: avm/res/fabric/capacity/main.bicep [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+resource fabricCapacities 'Microsoft.Fabric/capacities@2023-11-01' = {
+  name: name
+  location: location
+  tags: tags
+  sku: {
+    name: skuName
+    tier: skuTier
+  }
+  properties: {
+    // Required properties here
```
Comment (eriqua) [inline avm/res/fabric/capacity/main.bicep:79]:
```suggestion
```

File: avm/res/fabric/capacity/main.bicep [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+    administration: {
+      members: adminMembers
+    }
+    // Optional properties here
```
Comment (eriqua) [inline avm/res/fabric/capacity/main.bicep:83]:
```suggestion
```

File: avm/res/fabric/capacity/main.bicep [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+  }
+}
+
+// ============ //
+// Outputs      //
+// ============ //
+
+// Add your outputs here
```
Comment (eriqua) [inline avm/res/fabric/capacity/main.bicep:91]:
```suggestion
```

File: avm/res/fabric/capacity/main.bicep [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+@description('The resource ID of the deployed Fabric resource.')
+output resourceId string = fabricCapacities.id
+
+@description('The name of the deployed Fabric resource.')
+output name string = fabricCapacities.name
+
+@description('The location the resource was deployed into.')
+output location string = fabricCapacities.location
+
+// ================ //
```
Comment (eriqua) [inline avm/res/fabric/capacity/main.bicep:105]: Please remove this section if not used

File: avm/res/fabric/capacity/main.bicep [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+  'F32'
+  'F64'
+  'F128'
+  'F256'
+  'F512'
+  'F1024'
+  'F2048'
+])
+@description('Optional. SKU tier of the Fabric resource.')
+param skuName string = 'F2'
```
Comment (eriqua) [inline avm/res/fabric/capacity/main.bicep:28]: Is it intended to have a default value for this property?

File: avm/res/fabric/capacity/main.json [hunk 11]
```diff
@@ -0,0 +1,146 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+  "languageVersion": "2.0",
+  "contentVersion": "1.0.0.0",
+  "metadata": {
+    "_generator": {
+      "name": "bicep",
+      "version": "0.29.47.4906",
```
Comment (eriqua) [inline avm/res/fabric/capacity/main.json:8]: Not mandatory but just a heads up that a newer version of the bicep cli is available

File: avm/res/fabric/capacity/tests/e2e/defaults/main.test.bicep [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+metadata name = 'Using only defaults'
+metadata description = 'This instance deploys the module with the minimum set of required parameters.'
+
+// ========== //
+// Parameters //
+// ========== //
+
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+// e.g., for a module 'network/private-endpoint' you could use 'dep-dev-network.privateendpoints-${serviceShort}-rg'
```
Comment (eriqua) [inline avm/res/fabric/capacity/tests/e2e/defaults/main.test.bicep:12]:
```suggestion
```

File: avm/res/fabric/capacity/tests/e2e/defaults/main.test.bicep [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+param resourceGroupName string = 'dep-${namePrefix}-fabric-capacities-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to.')
+param resourceLocation string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+// e.g., for a module 'network/private-endpoint' you could use 'npe' as a prefix and then 'waf' as a suffix for the waf-aligned test
```
Comment (eriqua) [inline avm/res/fabric/capacity/tests/e2e/defaults/main.test.bicep:19]:
```suggestion
```

File: avm/res/fabric/capacity/tests/e2e/defaults/main.test.bicep [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+// Test Execution //
+// ============== //
+
+@batchSize(1)
+module testDeployment '../../../main.bicep' = [
+  for iteration in ['init', 'idem']: {
+    scope: resourceGroup
+    name: '${uniqueString(deployment().name, resourceLocation)}-test-${serviceShort}-${iteration}'
+    params: {
+      // You parameters go here
```
Comment (eriqua) [inline avm/res/fabric/capacity/tests/e2e/defaults/main.test.bicep:46]:
```suggestion
```

File: avm/res/fabric/capacity/tests/e2e/defaults/main.test.bicep [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+      name: '${namePrefix}${serviceShort}001'
+      location: resourceLocation
+      adminMembers: [
```
Comment (eriqua) [inline avm/res/fabric/capacity/tests/e2e/defaults/main.test.bicep:49]: This parameter is of type Optional. Defaults tests should only use required parameters. Either remove this parameter from the defaults test or, if needed, make it mandatory by removing the empty default value and changing the type in the parameter description

File: avm/res/fabric/capacity/tests/e2e/waf-aligned/main.test.bicep [hunk 16]
```diff
@@ -0,0 +1,54 @@
...
+metadata name = 'WAF-aligned'
+metadata description = 'This instance deploys the module in alignment with the best-practices of the Azure Well-Architected Framework.'
+
+// ========== //
+// Parameters //
+// ========== //
+
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+// e.g., for a module 'network/private-endpoint' you could use 'dep-dev-network.privateendpoints-${serviceShort}-rg'
```
Comment (eriqua) [inline avm/res/fabric/capacity/tests/e2e/waf-aligned/main.test.bicep:12]:
```suggestion
```

File: avm/res/fabric/capacity/tests/e2e/waf-aligned/main.test.bicep [hunk 17]
```diff
@@ -0,0 +1,54 @@
...
+param resourceGroupName string = 'dep-${namePrefix}-fabric-capacities-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to.')
+param resourceLocation string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+// e.g., for a module 'network/private-endpoint' you could use 'npe' as a prefix and then 'waf' as a suffix for the waf-aligned test
```
Comment (eriqua) [inline avm/res/fabric/capacity/tests/e2e/waf-aligned/main.test.bicep:19]:
```suggestion
```

File: avm/res/fabric/capacity/tests/e2e/waf-aligned/main.test.bicep [hunk 18]
```diff
@@ -0,0 +1,54 @@
...
+// Test Execution //
+// ============== //
+
+@batchSize(1)
+module testDeployment '../../../main.bicep' = [
+  for iteration in ['init', 'idem']: {
+    scope: resourceGroup
+    name: '${uniqueString(deployment().name, resourceLocation)}-test-${serviceShort}-${iteration}'
+    params: {
+      // You parameters go here
```
Comment (eriqua) [inline avm/res/fabric/capacity/tests/e2e/waf-aligned/main.test.bicep:46]:
```suggestion
```

File: avm/res/fabric/capacity/main.bicep
Comment (eriqua) [inline avm/res/fabric/capacity/main.bicep:1]: Any must have interface available for this resource (e.g. private endpoints, locks, rbac, managed identity, cmk, ...)?

File: .github/CODEOWNERS
Comment (eriqua) [inline .github/CODEOWNERS:1]: Please also add corresponding workflow yml

File: avm/res/fabric/capacity/main.bicep (hunk 1 above)
Comment (hundredacres) [reply avm/res/fabric/capacity/main.bicep:2]: Fixed.

File: avm/res/fabric/capacity/main.bicep (hunk 3 above)
Comment (hundredacres) [reply avm/res/fabric/capacity/main.bicep:31]: Updated. There's only 1 SKU that I know of for now.

File: avm/res/fabric/capacity/main.bicep (hunk 4 above)
Comment (hundredacres) [reply avm/res/fabric/capacity/main.bicep:40]: Fixed

File: avm/res/fabric/capacity/main.bicep (hunk 5 above)
Comment (hundredacres) [reply avm/res/fabric/capacity/main.bicep:67]: Fixed

File: avm/res/fabric/capacity/main.bicep (hunk 7 above)
Comment (hundredacres) [reply avm/res/fabric/capacity/main.bicep:83]: Fixed

File: avm/res/fabric/capacity/main.bicep (hunk 8 above)
Comment (hundredacres) [reply avm/res/fabric/capacity/main.bicep:91]: Fixed

File: avm/res/fabric/capacity/main.bicep (hunk 9 above)
Comment (hundredacres) [reply avm/res/fabric/capacity/main.bicep:105]: Fixed.

File: avm/res/fabric/capacity/main.bicep (hunk 10 above)
Comment (hundredacres) [reply avm/res/fabric/capacity/main.bicep:28]: Capacity is expensive, so wanted to default to the cheapest SKU. I can remove it if we want to make it a mandatory param.

File: avm/res/fabric/capacity/tests/e2e/defaults/main.test.bicep (hunk 12 above)
Comment (hundredacres) [reply avm/res/fabric/capacity/tests/e2e/defaults/main.test.bicep:12]: Fixed.

File: avm/res/fabric/capacity/tests/e2e/defaults/main.test.bicep (hunk 13 above)
Comment (hundredacres) [reply avm/res/fabric/capacity/tests/e2e/defaults/main.test.bicep:19]: Fixed

File: avm/res/fabric/capacity/tests/e2e/defaults/main.test.bicep (hunk 14 above)
Comment (hundredacres) [reply avm/res/fabric/capacity/tests/e2e/defaults/main.test.bicep:46]: Fixed.

File: avm/res/fabric/capacity/tests/e2e/waf-aligned/main.test.bicep (hunk 16 above)
Comment (hundredacres) [reply avm/res/fabric/capacity/tests/e2e/waf-aligned/main.test.bicep:12]: Fixed.

File: avm/res/fabric/capacity/tests/e2e/waf-aligned/main.test.bicep (hunk 17 above)
Comment (hundredacres) [reply avm/res/fabric/capacity/tests/e2e/waf-aligned/main.test.bicep:19]: Fixed.

File: avm/res/fabric/capacity/README.md (hunk 2 above)
Comment (hundredacres) [reply avm/res/fabric/capacity/README.md]: Fixed

File: .github/CODEOWNERS
Comment (hundredacres) [reply .github/CODEOWNERS:1]: Added workflow file.

File: avm/res/fabric/capacity/main.bicep
Comment (hundredacres) [reply avm/res/fabric/capacity/main.bicep:1]: In the future, but not now. Most folks would use PS scripts to finish deploying the additional components of a Fabric deployment.

File: avm/res/fabric/capacity/main.bicep (hunk 10 above)
Comment (donheerschap) [reply avm/res/fabric/capacity/main.bicep:28]: If I may, understood is that defaults in a module should be WAF Alligned, Considering workspace identity and managed private endpoints are a thing from F64 and therefore better candidate being WAF alligned I would suggest make this F64

Comment (hundredacres) [reply avm/res/fabric/capacity/main.bicep:28]: Happy to default to F64, but I'm still concerned about the cost though. F2 is ~ $250/mo, while a F64 is ~ $9300/mo. I wouldn't want someone deploying defaults and get hit with a multi thousand-dollar bill.

File: avm/res/fabric/capacity/main.bicep
Comment (eriqua) [reply avm/res/fabric/capacity/main.bicep:1]: This makes me a sad panda 🐼

File: avm/res/fabric/capacity/main.bicep (hunk 10 above)
Comment (eriqua) [reply avm/res/fabric/capacity/main.bicep:28]: Let me bring this up with the core team @hundredacres @donheerschap as I believe these are both good points. We use to prefer waf alignment over cost, but I'd like to double check if there should be a trade off in this case

File: avm/res/fabric/capacity/tests/e2e/waf-aligned/main.test.bicep (hunk 16 above)
Comment (eriqua) [reply avm/res/fabric/capacity/tests/e2e/waf-aligned/main.test.bicep:12]: Probably a missing push, I cannot see comments removed from this file

File: avm/res/fabric/capacity/main.bicep (hunk 10 above)
Comment (eriqua) [reply avm/res/fabric/capacity/main.bicep:28]: Meanwhile, if F64 is the waf compliant one, should that at least be the value for the waf-aligned test? Are workspace identity and managed private endpoint features we can integrate in the module?

File: avm/res/fabric/capacity/main.bicep [hunk 19]
```diff
@@ -0,0 +1,92 @@
...
+        telemetry: {
+          type: 'String'
+          value: 'For more information, see https://aka.ms/avm/TelemetryInfo'
+        }
+      }
+    }
+  }
+}
+
+resource fabricCapacities 'Microsoft.Fabric/capacities@2023-11-01' = {
```
Comment (eriqua) [inline avm/res/fabric/capacity/main.bicep:63]: As per convention I'd suggest a singular form for the symbolic name
```suggestion
resource fabricCapacity 'Microsoft.Fabric/capacities@2023-11-01' = {
```

File: avm/res/fabric/capacity/tests/e2e/waf-aligned/main.test.bicep (hunk 16 above)
Comment (hundredacres) [reply avm/res/fabric/capacity/tests/e2e/waf-aligned/main.test.bicep]: Pushed

File: avm/res/fabric/capacity/main.bicep (hunk 10 above)
Comment (hundredacres) [reply avm/res/fabric/capacity/main.bicep:28]: I have found no WAF documentation for Fabric that would provide guidance on the acceptable sku to use. Also there are no endpoint or identity features available atm.

Comment (donheerschap) [reply avm/res/fabric/capacity/main.bicep:28]: @eriqua There is [this request](https://github.com/Azure/bicep/issues/12905#issuecomment-2268310971) for fabric provider for bicep (as it's not available in ARM).

My suggestion would be either to put the F64 in the WAF e2e test, or make it default and create a minimal e2e test with F2 in it.

**Comment by ReneHezser** [comment]
@hundredacres 
We have added examples for Bicep parameter files to the Readme. This has been applied to all published modules but needs to be done for PRs as well. Can you please run the Set-AVMModule utility as detailed [here](https://azure.github.io/Azure-Verified-Modules/contributing/bicep/bicep-contribution-flow/generate-bicep-module-files/). It is required for the validation pipeline to succeed and the contribution to be published.

Please reach out if any support is needed.

File: avm/res/fabric/capacity/main.bicep (hunk 19 above)
Comment (hundredacres) [reply avm/res/fabric/capacity/main.bicep]: Updated

**Comment by eriqua** [comment]
> @hundredacres We have added examples for Bicep parameter files to the Readme. This has been applied to all published modules but needs to be done for PRs as well. Can you please update your branch and run the Set-AVMModule utility as detailed [here](https://azure.github.io/Azure-Verified-Modules/contributing/bicep/bicep-contribution-flow/generate-bicep-module-files/). It is required for the validation pipeline to succeed and the contribution to be published.
> 
> Please reach out if any support is needed.

@hundredacres all comments resolved and lgtm.
Please perform the above as a last step required before approval when you get the chance:
- pull latest from upstream
- regen Readme

Thanks for the great work! 💪

**Comment by hundredacres** [comment]
> > @hundredacres We have added examples for Bicep parameter files to the Readme. This has been applied to all published modules but needs to be done for PRs as well. Can you please update your branch and run the Set-AVMModule utility as detailed [here](https://azure.github.io/Azure-Verified-Modules/contributing/bicep/bicep-contribution-flow/generate-bicep-module-files/). It is required for the validation pipeline to succeed and the contribution to be published.
> > Please reach out if any support is needed.
> 
> @hundredacres all comments resolved and lgtm. Please perform the above as a last step required before approval when you get the chance:
> 
> * pull latest from upstream
> * regen Readme
> 
> Thanks for the great work! 💪

Done.

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**